### PR TITLE
Ecm 24 respondent rep multiple update

### PR DESCRIPTION
**Pull request by btensay**
Respondent rep update modification and logic added.

https://tools.hmcts.net/jira/browse/ECM-24

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/ecm/common/model/servicebus/tasks/UpdateDataTask.java [hunk 1]
```diff
@@ -204,51 +205,86 @@ private void updateJudgement(CaseData caseData, JudgementType judgementType) {
...
     }
 
-    private void updateRespondentRep(CaseData caseData, RepresentedTypeR representedType) {
-
-        //case 1: update is adding
-
-        //case 2: update is removing
+    private void updateRespondentRep(CaseData caseData, UpdateDataModel updateDataModel) {
+        RepresentedTypeR representedType = updateDataModel.getRepresentedType();
+        String isRespondentRepRemovalUpdate = updateDataModel.getIsRespondentRepRemovalUpdate();
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ecm/common/model/servicebus/tasks/UpdateDataTask.java:210]: Move this within the following if statement as you aren't using it outside the if statement

File: src/main/java/uk/gov/hmcts/ecm/common/model/servicebus/tasks/UpdateDataTask.java [hunk 2]
```diff
@@ -204,51 +205,86 @@ private void updateJudgement(CaseData caseData, JudgementType judgementType) {
...
                                             .equals(representedType.getRespRepName()))
                             .findAny();
 
-            if (respondentSumTypeItemOptional.isPresent()) {
+            boolean respondentSumTypeIsPresent = respondentSumTypeItemOptional.isPresent();
 
-                if (caseData.getRepCollection() != null) {
-
-                    boolean found = false;
+            if(respondentSumTypeIsPresent){
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ecm/common/model/servicebus/tasks/UpdateDataTask.java:224]: Minor thing but you might as well write
`if (respondentSumTypeItemOptional.isPresent()) {`

File: src/main/java/uk/gov/hmcts/ecm/common/model/servicebus/tasks/UpdateDataTask.java [hunk 3]
```diff
@@ -204,51 +205,86 @@ private void updateJudgement(CaseData caseData, JudgementType judgementType) {
...
 
-                    caseData.setRepCollection(
-                            new ArrayList<>(Collections.singletonList(createRespondentRepTypeItem(representedType))));
-                }
+        for (RepresentedTypeRItem representedTypeRItem : caseData.getRepCollection()) {
 
+            if (representedTypeRItem.getValue().getRespRepName() != null &&
+                    (representedTypeRItem.getValue().getRespRepName()
+                    .equals(representedType.getRespRepName()))) {
+                item = representedTypeRItem;
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ecm/common/model/servicebus/tasks/UpdateDataTask.java:270]: Can you return representedTypeRIItem here instead?

**Review by gauravchawla73463 — APPROVED** [review]

**Review by jthmcts — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/ecm/common/model/servicebus/tasks/UpdateDataTask.java [hunk 4]
```diff
@@ -204,51 +205,86 @@ private void updateJudgement(CaseData caseData, JudgementType judgementType) {
...
+    private void addRespondentRepRemovalUpdate(CaseData caseData, RepresentedTypeR representedType){
+        if(representedTypeRItemExists(caseData, representedType)){
+            getExistingRepresentedTypeRItem(caseData, representedType).setValue(new RepresentedTypeR());
+        }
+    }
 
-                    boolean found = false;
+    private void addRespondentRepUpdates(CaseData caseData, UpdateDataModel updateDataModel){
+        boolean caseDataHasRep = (caseData.getRepCollection() != null);
+        RepresentedTypeR representedType = updateDataModel.getRepresentedType();
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ecm/common/model/servicebus/tasks/UpdateDataTask.java:241]: representedType is only used in the following if statement so it should be declared in there too

File: src/main/java/uk/gov/hmcts/ecm/common/model/servicebus/tasks/UpdateDataTask.java [hunk 5]
```diff
@@ -204,51 +205,86 @@ private void updateJudgement(CaseData caseData, JudgementType judgementType) {
...
+                RepresentedTypeRItem representedTypeRItem = getExistingRepresentedTypeRItem(caseData,
+                        representedType);
+                if(representedTypeRItem != null)
+                {
+                    representedTypeRItem.setValue(representedType);
+                }
+            }
+            else{
+                caseData.getRepCollection().add(createRespondentRepTypeItem(representedType));
+            }
```
Comment (jthmcts) [inline src/main/java/uk/gov/hmcts/ecm/common/model/servicebus/tasks/UpdateDataTask.java:255]: Could you improve the formatting of this. There are two different styles being used for if statements.
See https://google.github.io/styleguide/javaguide.html
If particular https://google.github.io/styleguide/javaguide.html#s4.1.2-blocks-k-r-style

```
if (value) {
   stmt;
} else {
   stmt;
}
```

Comment (btensay) [reply src/main/java/uk/gov/hmcts/ecm/common/model/servicebus/tasks/UpdateDataTask.java]: done

**Review by jthmcts — APPROVED** [review]

**Pull request merged**